frontend/catalyst/pennylane_extensions.py

```diff
@@ -935,9 +945,21 @@ def check_is_iterable(x, hint):
         fn = _ensure_differentiable(f)
         grad_params = _check_grad_params(method, scalar_out, h, argnum)
 
-        jaxpr, _ = _make_jaxpr_check_differentiable(fn, grad_params, *params)
+        jaxpr, out_tree = _make_jaxpr_check_differentiable(fn, grad_params, *params)
 
         results = vjp_p.bind(*params, *cotangents, jaxpr=jaxpr, fn=fn, grad_params=grad_params)
+
+        if out_tree.children() != []:
+            func_res = results[: len(jaxpr.out_avals)]
+            vjps = results[len(jaxpr.out_avals) :]
+            func_res = tree_unflatten(out_tree, func_res)
+            # We do not change the shape of the VJP as it is the same as the parameters not
+            # as the output
+            results = tuple([func_res, tuple(vjps)])
```

Comment:
I was considering if the vjps should be unflattened with the function argument trees, but probably not? 😅

Reply:
Well the cotangent arguments actually

Reply:
The VJP should have the same shape as the parameters it is correct but the goal here was to return pytrees not to have them as arguments.

Reply:
Did you plan on adding PyTree argument support in a separate PR? I think in this case tackling them together would be effective due to how the cotangent and gradient results are linked for example.

Reply:
I can do that in another PR, but it was not part of this quarter goals. (Happy to add it to shorcut)

Reply:
done in #549